### ISSUE TITLE
Review the "helpful links" on the right

### ISSUE THREAD

**Issue by PaulPoulain** [coral-erm/coral#35]
The "helpful link" area, on the right:
- takes a lot of space
- has duplicate informations

for example: http://coraldemo.library.tamu.edu/resources/resource.php?resourceID=75
![capture du 2016-05-03 13 57 58](https://cloud.githubusercontent.com/assets/4696271/14982492/170ab4c2-1137-11e6-844f-92b2f1fe8c6d.png)
- the "child records" is also displayed as Related products in the main area
- the Organization module: Ebrary is also displayed as "Organizations: "

**Comment by PaulPoulain** [comment]
I added the label "bug" because:
- there are duplicate SQL queries that have an impact on performance
-  when organization or child record is long, the display can become quite hard to read

**Comment by techsvcslib** [comment]
I disagree that this is a bug. A bug is when a feature breaks. This isn’t a break, it’s just something that could perhaps be done better.

Steve

**Comment by PaulPoulain** [comment]
While I agree it's not a critical problem, I still think it's a bug, and not en enhancement. We've loaded large packages, with a lot of titles, and the performances are awful & the display ugly.
Here is a screenshot:
![capture du 2016-05-04 17 56 40](https://cloud.githubusercontent.com/assets/4696271/15020321/d8e7f7e0-1221-11e6-9327-91f9bbdb19be.png)

 * you can see there are duplicated informations
 * they've different labels (Related product: and Child Record(s) ), that's hard to understand for users.
 * on a DB with packages having x00 of title, the performance hit is really huge. We've removed the "helpful link" part on a sample DB with 230 000 resources, and it's obvious !

**Comment by techsvcslib** [comment]
Paul,

I still strongly disagree that these are bugs. I’m not trying to be difficult about it, and believe me, I am fully aware of the aspects you raise. But I think this is an important distinction for us in the SC to make and to have a shared understanding about (what is a bug vs. an enhancement).

Steve

**Comment by PaulPoulain** [comment]
Le 03/05/2016 19:21, techsvcslib a écrit :
> I disagree that this is a bug. A bug is when a feature breaks. This 
> isn’t a break, it’s just something that could perhaps be done better.
We can discuss for years of this topic I think. Most project solve this 
by having different levels of bugs (blocking, critical, major, minor, 
aesthetic,...) or having an other flag.

I've added many issues to github tracker, that you'll not consider as 
bugs probably, but, for me, they are bugs:
https://github.com/Coral-erm/Coral/issues/36 => it's a design issue
https://github.com/Coral-erm/Coral/issues/29 => nothing is "broken", you 
can use Coral. But I consider the behavior as illogical.

Even for https://github.com/Coral-erm/Coral/issues/37, I have hesitated 
to classify it as "bug" : it's a design issue like the #36 one.

I'd like to talk about this in a future meeting (will add the topic to 
trello immediately)

-- 
Paul Poulain, Associé-gérant / co-owner
BibLibre, Services en logiciels libres pour les bibliothèques
BibLibre, Open Source software and services for libraries

**Comment by techsvcslib** [comment]
Paul,

I agree — we can argue about this for a while. Let’s plan to talk about this as a group at some point (the general issue of “bug” vs. “enhancement”).

Steve

**Comment by benheet** [comment]
I would agree that the performance issues caused by trying to display hundreds/thousands of child record titles would be a bug. The existence of the helpful links section and the fact the same information is displayed in multiple locations of the same record is not a bug. We might be able to find agreement by being more specific in what we are pointing out as the feature in question.

**Comment by benheet** [comment]
Can we craft a list of the entirety of things displayed in the helpful links section? Is it just print view, child records, organizations? I would be open to a conversation about other ways to provide this feature. For example, print view could be placed somewhere else within the display. Child resources, when the list gets long, could be linked to with something like a 'more children' link that could send the user to the search results page with results scoped to children of the selected record, and I could live without a second link to the related organizations.

**Comment by techsvcslib** [comment]
I think there is much more to the specific issue of display of child records than the performance issue. I still disagree that display of hundreds/thousands of child records causes performance issues and that this should be classified as a bug. By the way, we’ve locally dealt with this for at least three years as one of the few libraries who have been actively using parent/child relationships for large packages for a while, and there isn’t a huge performance problem with it in our instance.

Much more problematic than the performance issue is the actual display of the children. It’s not in any sensible order (e.g. alphabetical), and overall the display is way too long and cumbersome to use when hundreds of links get displayed. I’d prefer something more manageable like a link to export a title list of all children, or something like that.

Steve

**Comment by techsvcslib** [comment]
I like the idea for “More children” that you propose, Ben. Another thing that gets displayed in the helpful links is a link to the license. That’s also available within the Acquisitions tab so I’m ok with that not being in helpful links any more.

Steve

**Comment by xsong9** [comment]
This looks like a change that needs further discussion. Let's target it for a later release.

**Comment by rtatterson** [comment]
Since this issue is tagged for an older version of CORAL, closing this ticket.
